pydantic/types.py

```diff
@@ -990,6 +990,10 @@ def __get_pydantic_core_schema__(
                     function=_validators.import_string, schema=handler(source), serialization=serializer
                 )
 
+        @classmethod
+        def __get_pydantic_json_schema__(cls, cs: CoreSchema, handler: GetJsonSchemaHandler) -> JsonSchemaValue:
+            return core_schema.str_schema() if (cls is source) else handler(core_schema.str_schema())
```

Comment:
```suggestion
            return handler(core_schema.str_schema())
```

Reply:
Ah my apologies, you were right the first time! We're dealing with json schema so your approach is correct :)